pkg/operator/ceph/cluster/mon/service.go

```diff
@@ -52,6 +52,18 @@ func (c *Cluster) createService(mon *monConfig) (string, error) {
 		svcDef.Spec.ClusterIP = v1.ClusterIPNone
 	}
 
+	// allows operators the ability to expose their rook-ceph mons externally
+	// by supplying their own `loadBalancerIP`. Since this field can only be used
+	// with `type` LoadBalancer, we automatically convert the service definition to use
+	// `LoadBalancer` instead of `ClusterIP` (the default)
+	if c.spec.Network.LoadBalancerService {
+		svcDef.Spec.Type = v1.ServiceTypeLoadBalancer
+
+		if len(c.spec.Network.LoadBalancerIP) > 0 {
+			svcDef.Spec.LoadBalancerIP = c.spec.Network.LoadBalancerIP
+		}
+	}
```

Comment:
If it's of type LoadBalancer, will the ClusterIP not be set? At the end of this method, the spec.ClusterIP is returned to the caller. Would we need to return LoadBalancerIP instead?